frontend/src/metabase/components/ErrorDetails/ErrorDetails.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useState } from "react";
+import { t } from "ttag";
+import cx from "classnames";
+
+import { ErrorDetailsProps } from "./types";
+
+export default function ErrorDetails({
+  details,
+  centered,
+  className,
+}: ErrorDetailsProps) {
+  const [showError, setShowError] = useState(false);
+
+  if (!details) {
+    return null;
+  }
+
+  function toggleShowError() {
+    setShowError(showError => !showError);
+  }
+
+  return (
+    <div className={className}>
+      <div className={centered ? "text-centered" : "text-left"}>
+        <a onClick={toggleShowError} className="link cursor-pointer">
+          {showError ? t`Hide error details` : t`Show error details`}
```

Comment:
Here's the changed logic.
- From setting `showError` to `true` 👉  Toggling `showError`
- From always showing `Show error details` 👉  Showing either `Hide error details` or `Show error details` based on the `showError` flag.